MAUI/DataForm/Data-annotations.md

```diff
@@ -0,0 +1,475 @@
+    public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight ResourceType attribute}
+
+   -----------------------
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight Prompt attribute}
+
+    [Display(Prompt = "Enter your name")]
+    public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight Order attribute}
+
+    [Display(Order =2)]
+    public string MiddleName { get; set; }
+
+    [Display(Order =1)]
+    public string LastName { get; set; }
+
+    [Display(Order =0)]
+    public string FirstName { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+
+
+## Validation attribute
+
+<table>
+<tr>
+<th>
+Property
+</th>
+<th>
+Details
+</th>
+</tr>
+<tr>
+<td>
+{{'[MinLength](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.minlengthattribute?view=net-7.0)'| markdownify }}
+</td>
+<td>
+Specifies the required minimum length. Refer to `Link` for more details.
+</td>
+</tr>
+<tr>
+<td>
+{{'[MaxLength](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.maxlengthattribute?view=net-7.0)'| markdownify }}
+
+</td>
+<td>
+Specifies the required maximum length. Refer to `Link` for more details.
+</td>
+</tr>
+<tr>
+<td>
+{{'[Required](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.requiredattribute?view=net-7.0)'| markdownify }}
+</td>
+<td>
+Specifies the required data field value. Refer to `Link` for more details.
+</td>
+</tr>
+<tr>
+<td>
+{{'[StringLength](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.stringlengthattribute?view=net-7.0)'| markdownify }}
+</td>
+<td>
+Specifies the required string length. Refer to `Link` for more details.
+</td>
+</tr>
+</table>
+
+{% tabs %}
+{% highlight MinLength attribute}
+
+   [MinLength(5,ErrorMessage ="Password length must be greater than 5 characters")]
+    public string Password { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight MaxLength attribute}
+
+    [MaxLength(20,ErrorMessage ="Maximum password length should be less than 20")]
+    public string Password { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight Required attribute}
+
+    [Required(AllowEmptyStrings =false, ErrorMessage ="Name is required")]
+    public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight StringLength attribute}
+
+    [StringLength(20,ErrorMessage ="Name should not exceed 15 characters")]
+    public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Bindable attribute
+
+It specifies whether the field should be auto generated or not. Refer to `Link`  for more details.
+
+## Editable attribute
+
+It specifies whether the data field is editable or not.
+
+## ReadOnly attribute
+
+It specifies whether the data field is read only or not. Refer to `Link` for more details.
```

Comment:
Refer to `Link` for more details. => remove

[ReadOnly(true)]
public string Name { get; set; }